### ISSUE TITLE
Travis-CI jobs on queue for a long time

### ISSUE THREAD

**Issue by AyanSinhaMahapatra** [nexB/scancode-toolkit#1790]
### Description

Travis-CI seems to be stuck at "Job recieved" for a long time before starting jobs across all nexB repositories. 

Quoting @majurg from aboutcode/discuss at Gitter

> @pombredanne Would it make sense to move away from Travis and towards azure pipelines only? Travis seems to be getting worse performance-wise, particularly for their unpaid offering.
The macOS side of Travis is awful and has been for some time.
IMO if we lighten up the load for Travis specifically just to get the tests to run in a timely manner, what is the point of having it as a valid CI check? 
Also, the run-times reported by Travis are not the whole story either; Some jobs are staying in the queue for a very long time before they are even started (upwards of 30 minutes in some cases). This is especially bad in macOS cases. And this stalled queue time is not reported at all by Travis.
Just to add, the Travis CI job for JUST the branch on nexB/scancode-toolkit#1741 has taken over 2.5 hours so far and is still waiting to complete.
Note this is JUST the branch; there is another Travis CI job that is taking just as long (if not longer) for the PR as well.

And I'm also facing similar issues, in nexB/aboutcode. It takes significantly longer for some reason, stuck at Job received. I have a Travis-CI setup for my fork of aboutcode and that always finishes in a couple of minutes.

**Comment by pombredanne** [comment]
Travis used to be awesome... but it is a tad less so lately.
What I think we will do is to limit more and more what we run there only to ruhttps://github.com/nexB/scancode-toolkit/blob/develop/tests/licensedcode/licensedcode_test_utils.py#L198n basic smoke tests on a few more OSes.

**Comment by melroy89** [comment]
Its still a problem in 2020...

**Comment by jtwebman** [comment]
Yep, it is so bad I am moving my open source projects to GitHub actions.

**Comment by melroy89** [comment]
I did the same with another project (also to github actions): https://github.com/vasild/cpp-ipfs-http-client/pull/38

So, if you need some inspiration: https://github.com/vasild/cpp-ipfs-http-client/blob/master/.github/workflows/build_and_test.yml

**Comment by Nic30** [comment]
Does anyone know the reason why travis-ci become so incredibly slow? My jobs are spending more than 4 hours in the queue at average (job itself takes 1-8 min depending on project).

(Edit.) Found this: https://www.theregister.com/2020/11/02/travis_ci_pricng/
(Edit2) https://www.traviscistatus.com/#month  396 running 7.5k-2.5K jobs in queue constantly

**Comment by melroy89** [comment]
Yea, they claim they are migration from the .org to .com, causing way too much impact I think.

Also I'm afraid they are pushing us to take a subscription.

**Comment by AyanSinhaMahapatra** [comment]
I see the prompt `Please be aware travis-ci.org will be shutting down in several weeks, with all accounts migrating to travis-ci.com. Please stay tuned here for more information.` is scancode travis builds and @sbs2001 also mentioning at gitter@discuss that we should migrate too (given there aren't any hidden implications/non-free conditions).

Should we make this migration and/or Github actions, because Travis now seems to only be giving a certain fixed number of VMs for a repository and that's taking too much time.

**Comment by pombredanne** [comment]
I am dropping Travis entirely for now

**Comment by jtwebman** [comment]
I moved everything to Github actions

**Comment by sritasngh** [comment]
Hello everyone!
I found this discussion interesting. It would be great if we migrate ScanCode toolkit to GitHub Action. I have gone through [Azure pipelines](https://github.com/nexB/scancode-toolkit/blob/develop/azure-pipelines.yml) , [Appveyor](https://github.com/itssingh/scancode-toolkit/blob/develop/appveyor.yml), [Travis](https://github.com/itssingh/scancode-toolkit/blob/develop/.travis.yml) and other CIs. I found that migration will required time and this is something which I would like to do in the summer. I want to ask @pombredanne @AyanSinhaMahapatra if I can propose this idea for GSoC-21.

**Comment by AyanSinhaMahapatra** [comment]
@itssingh don't think this will take a lot of time, see https://github.com/nexB/scancode.io/pull/80 for example. So this certainly would not be a project that fits GSoC, neither in terms of complexity nor in terms of time required, IMHO. Also don't think anything would replace the azure pipelines currently.

We tested our documentation only through Travis, so I think dropping Travis also dropped doc checks out of CI, it could make sense to add GitHub actions for that similarly. @pombredanne would be able to say more on that.

**Comment by sritasngh** [comment]
@AyanSinhaMahapatra okay. I would like to work on adding Github actions to migrate Travis CI in the coming time.

**Comment by steven-esser** [comment]
@itssingh We have already migrated to using azure pipelines and travis support will drop soon (if it hasn't already). I would find another issue to work on.

**Comment by pombredanne** [comment]
we no longer use travis.
